### ISSUE TITLE
Error Fitting Terga2

### ISSUE THREAD

**Issue by berunge** [predomics/predomicspkg#2]
I am encountering an error trying to create models with terga2. I was able to reproduce the error by running the vignette code, which I have reproduced the relevant portion of below. I am running R version 4.0.2.

``` r
# load the library
library(predomics)
#> Loading required package: doSNOW
#> Warning: package 'doSNOW' was built under R version 4.0.3
#> Loading required package: foreach
#> Warning: package 'foreach' was built under R version 4.0.3
#> Loading required package: iterators
#> Warning: package 'iterators' was built under R version 4.0.3
#> Loading required package: snow
#> Warning: package 'snow' was built under R version 4.0.3
#> Warning: replacing previous import 'gtools::na.replace' by 'glmnet::na.replace'
#> when loading 'predomics'
#> Warning: replacing previous import 'ggplot2::alpha' by 'kernlab::alpha' when
#> loading 'predomics'
#> Warning: replacing previous import 'ggplot2::margin' by 'randomForest::margin'
#> when loading 'predomics'
#> Warning: replacing previous import 'gridExtra::combine' by
#> 'randomForest::combine' when loading 'predomics'
library(ggplot2)
library(gridExtra)
library(pROC)
#> Type 'citation("pROC")' for a citation.
#> 
#> Attaching package: 'pROC'
#> The following objects are masked from 'package:stats':
#> 
#>     cov, smooth, var
library(reshape2)

# load the data
data("cir_train")
str(cir_train, max.level = 1)
#> List of 2
#>  $ X:'data.frame':   1045 obs. of  181 variables:
#>  $ y: num [1:181] 1 1 1 1 1 1 1 1 1 1 ...

# Filter the non informative variables
X <- cir_train$X; y <- cir_train$y # set global variables
X <- X[rowSums(X)!=0,]; dim(X) # filter out variables with only zero values
#> [1] 556 181
X <- filterNoSignal(X = X, side = 1, threshold = "auto", verbose = FALSE); dim(X) 
#> [1] 462 181

# Terga2
clf <- terga2(nCores = 1, 
              seed = 1, 
              plot = TRUE
)

clf$experiment$description <- "PLACEHOLDER"
res_clf <- fit(X = X, y = y, clf = clf, cross.validate = TRUE, nfolds = 2); # class(res_clf)
#> ... Database X is not a matrix! Converting ...
#> ... Classification mode, computing factor(y) for speedup and robustness
#> ... Computing feature correlation for speedup
#> ... Storing data in the classifier object for speedup
#> ... Computing ternary coefficients for speedup
#> ... One seed found, setting by default
#> ... Running the classifier terga2 with a single CPU
#> ... Second and faster version of terga fitting based on Genetic Algorithm heuristics ...
#> ... Cross validation mode
#> ... Starting cross validation not in parallel
#> ....................................................................................................
#> Error in if (class(pop.dense.noz) == "numeric") { : 
#>   the condition has length > 1
#> ....................................................................................................
#> Error in if (class(pop.dense.noz) == "numeric") { : 
#>   the condition has length > 1
#> ....................................................................................................
#> Error in if (class(pop.dense.noz) == "numeric") { : 
#>   the condition has length > 1
#> [1] "Error in if (class(pop.dense.noz) == \"numeric\") { : \n  the condition has length > 1\n"
#> attr(,"class")
#> [1] "try-error"
#> attr(,"condition")
#> <simpleError in if (class(pop.dense.noz) == "numeric") {    pop.dense.noz <- as.matrix(as.data.frame(pop.dense.noz))}: the condition has length > 1>
#> [1] "Error in if (class(pop.dense.noz) == \"numeric\") { : \n  the condition has length > 1\n"
#> attr(,"class")
#> [1] "try-error"
#> attr(,"condition")
#> <simpleError in if (class(pop.dense.noz) == "numeric") {    pop.dense.noz <- as.matrix(as.data.frame(pop.dense.noz))}: the condition has length > 1>
#> Error: $ operator is invalid for atomic vectors
```

<sup>Created on 2021-03-31 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by berunge** [comment]
I was able to look into this some more. The error seems to occur on either line 3279 or line 3300 in the global.lib code during a call for feature importance computations. I am able to run models by setting compute.importance = FALSE in the fit function.

**Comment by eprifti** [comment]
thank you @berunge for this. I'll update the code to take into account the matrix issue in the next few days. 
Best wishes!

**Comment by eprifti** [comment]
Thanks again @berunge. Fixed this in the 1.0.3 version.
